### DISCUSSION TITLE
[Solved] CastKodi doesn't connect - despite remotecontrol being enabled in kodi

### DISCUSSION THREAD

**Discussion by Killom** [regseb/castkodi#143] [General]
So the point in Cast Kodi OSD keeps bouncing left to right, without connecting to Kodi. It used to work some days ago. And I've changed nothing™ so far. Possible to debug this scenario somehow and identify the culprit?

Edit:
Found the "Debug this Add-On button" in FF. Well culprit might be HSTS; Because FF enforces SSL even for ws:// protocol

`Firefox cannot connect to the server at wss://kodi.box:9090/jsonrpc`

Edit2:
Also doesn't work, when Enforce HTTPS is disabled in FF

Edit3:
Connecting via telnet to kodi.box:9090 is working. Opening http://kodi.box:9090 in FF tab will yield JSON on status updates.

So, I've genuinely have no idea whats the issue

**Comment by regseb** [comment]
- What address do you enter in the Cast Kodi options?
- Is Kodi / _Settings_ / _Services_, what value do you have in:
  - _Allow remote control via HTTP_?
    - _Port_?
    - _Require authentication_?
    - _Enable SSL_?
  - _Allow remote control from applications on this system_?
  - _Allow remote control from applications on other systems_?
- In [`http://kodi.box/#settings/web`](http://kodi.box/#settings/web) (you may need to _Allow remote control via HTTP_ in Kodi), what value do you have in _Settings_ / _Web Interface_ / _Advanced options_: _Websockets port_ and _Websockets host_?

---

For me, with Firefox 130 and Kodi 21.1 (on OSMC 2024.08-1):

- When I fill `ws://rpi4.home:9090/jsonrpc` in the options form: Cast Kodi works correctly.
- When I fill `wss://rpi4.home:9090/jsonrpc` in the options form: I get the error message _Address of Kodi Web server wss://rpi4.home:9090/jsonrpc is invalid or Kodi's remote control isn't enabled._
- When I open the URL `http://rpi4.home:9090` directly in Firefox, the page loads indefinitely.

In Kodi / _Settings_ / _Services_:
- _Allow remote control via HTTP_: enabled
  - _Port_: `80`
  - _Require authentication_: enable
  - _Enable SSL_: disabled
- _Allow remote control from applications on this system_: enabled
- _Allow remote control from applications on other systems_: enabled

In `http://rpi4.home/#settings/web`:
- _Websockets port_: `9090`
- _Websockets host_: `rpi4.home`

**Reply by Killom** [reply]
Ok, I've managed to do a cross check on my Surface which turned out OK. So Its something device specific on my PC that went south here. Maybe Firefox Profile is damaged ... 

Edit:
Yes, its the FF profile. Guess it's not worth the time, digging any deeper here. Will just make a new one.

/solved

**Comment by Killom** [comment]
Done a revisit on this today, since I finally opted to not want to redo my FF profile:

Issue indeed was HTTPS Enforce mode. Maybe I forgot to restart FF after disabling when testing the last time. 

Enforce Mode enforces use of wss:// instead of ws:// - so exception for IP / FQDN of Kodi is needed if you don't want to go without enforce mode in general, since Kodi is not capable of WSS
